src/library/clementine/clementineplaylistmodel.cpp

```diff
@@ -0,0 +1,299 @@
+#include "library/clementine/clementineplaylistmodel.h"
+
+#include <QtDebug>
+
+#include "library/clementine/clementinedbconnection.h"
+#include "library/queryutil.h"
+#include "library/starrating.h"
+#include "library/trackcollectionmanager.h"
+#include "mixer/playermanager.h"
+#include "track/beatfactory.h"
+#include "track/beats.h"
+
+#define Clementine_TABLE "Clementine"
+#define CLM_VIEW_ORDER "position"
+#define CLM_ARTIST "artist"
+#define CLM_TITLE "title"
+#define CLM_DURATION "duration"
+#define CLM_URI "location"
+#define CLM_ALBUM "album"
+#define CLM_ALBUM_ARTIST "album_artist"
+#define CLM_YEAR "year"
+#define CLM_RATING "rating"
+#define CLM_GENRE "genre"
+#define CLM_GROUPING "grouping"
+#define CLM_TRACKNUMBER "tracknumber"
+#define CLM_DATEADDED "datetime_added"
+#define CLM_BPM "bpm"
+#define CLM_BITRATE "bitrate"
+#define CLM_COMMENT "comment"
+#define CLM_PLAYCOUNT "timesplayed"
+#define CLM_COMPOSER "composer"
+#define CLM_PREVIEW "preview"
+
+ClementinePlaylistModel::ClementinePlaylistModel(QObject* pParent,
+        TrackCollectionManager* pTrackCollectionManager,
+        ClementineDbConnection* pConnection)
+        : BaseSqlTableModel(pParent,
+                  pTrackCollectionManager,
+                  "mixxx.db.model.Clementine_playlist"),
+          m_pConnection(pConnection),
+          m_playlistId(-1) {
+}
+
+ClementinePlaylistModel::~ClementinePlaylistModel() {
```

Comment:
`= default;`

Reply:
added this in https://github.com/mixxxdj/mixxx/pull/3173/commits/f2a59de119ecefa97cf8d9ae8821a79d37650a05